### PR TITLE
chore(Poetry): Group dev deps with updated syntax

### PR DESCRIPTION
**Pull request by mwarres**
Previously, development dependencies were grouped under `[tool.poetry.dev-dependencies]` in `pyproject.toml`. This syntax will soon be deprecated. Group development dependencies under `[tool.poetry.group.dev.dependencies]`, the updated syntax as of Poetry v1.2.0.

### CODE REVIEW

**Review by Kurt-von-Laven — CHANGES_REQUESTED** [review]
Commit message: the [official Poetry docs](https://python-poetry.org/docs/managing-dependencies/#dependency-groups) say that the old syntax will be deprecated soon (but is not currently). Worth noting that the new syntax was released in Poetry 1.2.0.

**Comment by mwarres** [comment]
@Kurt-von-Laven Are you able to take a look at this and see if it's ok?

**Review by Kurt-von-Laven — APPROVED** [review]
**LGTM**

File: poetry.lock [hunk 1]
```diff
@@ -183,13 +183,13 @@ testing = ["covdefaults (>=2.3)", "coverage (>=7.3)", "diff-cover (>=7.7)", "pyt
 
 [[package]]
 name = "identify"
-version = "2.5.27"
+version = "2.5.28"
```
Comment (Kurt-von-Laven) [inline poetry.lock:186]: Did you run `poetry update` locally? Please remove unrelated changes from this pull request.

**Review by Kurt-von-Laven — CHANGES_REQUESTED** [review]

File: poetry.lock (hunk 1 above)
Comment (mwarres) [reply poetry.lock:186]: Just `poetry lock` and `poetry install`. I'll remove the change though. Thanks for catching this.

Comment (Kurt-von-Laven) [reply poetry.lock:186]: `poetry lock` runs `poetry update`, which is why [our pre-commit hook instead runs `poetry lock --no-update`](https://github.com/ScribeMD/pre-commit-hooks#poetry-lock) to prevent it from constantly performing lock file maintenance.

Comment (Kurt-von-Laven) [reply poetry.lock]: Still the same problem as before.

Comment (mwarres) [reply poetry.lock]: Oh darn. Sorry, just seeing this.

Comment (Kurt-von-Laven) [reply poetry.lock]: Same problem again.

**Review by Kurt-von-Laven — APPROVED** [review]
**LGTM**
